README.md

```diff
@@ -4,41 +4,209 @@ Please edit this README.md file with your answers to the worksheet questions.
+| 1 | 1 | 0     | 0     | 0              |
 
 ### c
+| A | B | C | A AND B | A AND C | (A AND B) OR (A AND C) |
+|---|---|---|---------|---------|------------------------|
+| 0 | 0 | 0 | 0       | 0       | 0                      |
+| 0 | 0 | 1 | 0       | 0       | 0                      |
+| 0 | 1 | 0 | 0       | 0       | 0                      |
+| 0 | 1 | 1 | 0       | 0       | 0                      |
+| 1 | 0 | 0 | 0       | 0       | 0                      |
+| 1 | 0 | 1 | 0       | 1       | 1                      |
+| 1 | 1 | 0 | 1       | 0       | 1                      |
+| 1 | 1 | 1 | 1       | 1       | 1                      |
+
+| A | B | C | B OR C | A AND (B OR C) |
+|---|---|---|--------|----------------|
+| 0 | 0 | 0 | 0      | 0              |
+| 0 | 0 | 1 | 1      | 0              |
+| 0 | 1 | 0 | 1      | 0              |
+| 0 | 1 | 1 | 1      | 0              |
+| 1 | 0 | 0 | 0      | 0              |
+| 1 | 0 | 1 | 1      | 1              |
+| 1 | 1 | 0 | 1      | 1              |
+| 1 | 1 | 1 | 1      | 1              |
 
 ### d
+| A | B | C | A OR B | A OR C | (A OR B) AND (A OR C) |
+|---|---|---|--------|--------|-----------------------|
+| 0 | 0 | 0 | 0      | 0      | 0                     |
+| 0 | 0 | 1 | 0      | 1      | 0                     |
+| 0 | 1 | 0 | 1      | 0      | 0                     |
+| 0 | 1 | 1 | 1      | 1      | 1                     |
+| 1 | 0 | 0 | 1      | 1      | 1                     |
+| 1 | 0 | 1 | 1      | 1      | 1                     |
+| 1 | 1 | 0 | 1      | 1      | 1                     |
+| 1 | 1 | 1 | 1      | 1      | 1                     |
+
+| A | B | C | B AND C | A OR (B AND C) |
+|---|---|---|---------|----------------|
+| 0 | 0 | 0 | 0       | 0              |
+| 0 | 0 | 1 | 0       | 0              |
+| 0 | 1 | 0 | 0       | 0              |
+| 0 | 1 | 1 | 1       | 1              |
+| 1 | 0 | 0 | 0       | 1              |
+| 1 | 0 | 1 | 0       | 1              |
+| 1 | 1 | 0 | 0       | 1              |
+| 1 | 1 | 1 | 1       | 1              |
 
 ## Question 4
 
 ### a
+| a | b | a AND b | NOT(a AND b) |
+|---|---|---------|--------------|
+| 0 | 0 | 0       | 1            |
+| 0 | 1 | 0       | 1            |
+| 1 | 0 | 0       | 1            |
+| 1 | 1 | 1       | 0            |
+
+| a | b | NOT a | NOT b | NOT a OR NOT b |
+|---|---|-------|-------|----------------|
+| 0 | 0 | 1     | 1     | 1              |
+| 0 | 1 | 1     | 0     | 1              |
+| 1 | 0 | 0     | 1     | 1              |
+| 1 | 1 | 0     | 0     | 0              |
 
 ### b
+The first program is to output "Hello" if [x] is an integer greater than 7 OR if [x] is a float greater than 7, while the second is to output "Hello" if [x] is an interger OR a float that is greater than 7. These mean the same thing because if we let [x] is an interger be [x = 1] and [x] is greater than 7 be [s = 1] then [(x AND s) OR (NOT x AND s)] is the same as [(x OR NOT x) AND s].
```

Comment:
Along the right lines -- however you are assuming that if `x` is not an integer then it must be a float. This might not be true -- need to treat the two as separate boolean conditions